### ISSUE TITLE
Solaris 10: Build fails in libedit with error: ‘ED_UNASSIGNED’ undeclared

### ISSUE THREAD

**Issue by quanah** [heimdal/heimdal#662]
**Describe the bug**
Build on Solaris 10 with gcc.  I hit failure with:

```
/export/home/build/sold/heimdal/lib/libedit/src/chared.c: In function ‘ch_init’:
/export/home/build/sold/heimdal/lib/libedit/src/chared.c:424:29: error: ‘ED_UNASSIGNED’ undeclared (first use in this function)
  el->el_chared.c_redo.cmd = ED_UNASSIGNED;
                             ^
/export/home/build/sold/heimdal/lib/libedit/src/chared.c:424:29: note: each undeclared identifier is reported only once for each function it appears in
/export/home/build/sold/heimdal/lib/libedit/src/chared.c: In function ‘ch_reset’:
/export/home/build/sold/heimdal/lib/libedit/src/chared.c:476:26: error: ‘ED_UNASSIGNED’ undeclared (first use in this function)
  el->el_state.lastcmd  = ED_UNASSIGNED;
                          ^
/export/home/build/sold/heimdal/lib/libedit/src/chared.c: In function ‘ch_end’:
/export/home/build/sold/heimdal/lib/libedit/src/chared.c:586:29: error: ‘ED_UNASSIGNED’ undeclared (first use in this function)
  el->el_chared.c_redo.cmd = ED_UNASSIGNED;
                             ^
make[1]: *** [Makefile:470: chared.lo] Error 1
make[1]: Leaving directory '/export/home/build/sold/SunOS10-i86pc_64/build/heimdal.i86pc_64/lib/libedit/src'
make: *** [Makefile:349: all] Error 2
```

Still working on the fix for this.

**Comment by quanah** [comment]
Duplicate of #309 which never got pushed into the current release series.

**Comment by quanah** [comment]
Actually this is a different problem related to the locale of the system

**Comment by quanah** [comment]
I found an obscure reference to this in an ancient thread, and the patch suggested there allowed the build to proceed.

```
diff --git a/lib/libedit/src/makelist b/lib/libedit/src/makelist
index 315d91e..264bfb5 100644
--- a/lib/libedit/src/makelist
+++ b/lib/libedit/src/makelist
@@ -41,6 +41,9 @@ then
 fi
 USAGE="Usage: $0 -h|-fc|-fh|-bh <filenames>"

+LC_ALL=C; LANG=C
+export LC_ALL LANG
+
 if [ "x$1" = "x" ]
 then
     echo $USAGE 1>&2
```

Reference: https://sourceforge.net/p/openipmi/discussion/112995/thread/198c3959/

**Comment by quanah** [comment]
The LANG setting on the system I'm building on is:

LANG=en_US.UTF-8

**Comment by lhoward** [comment]
Can't reproduce on Solaris 11 with gcc. What's in fcns.h? I have –

```c
...
#define ED_SEQUENCE_LEAD_IN              25
#define ED_START_OVER                    26
#define ED_TRANSPOSE_CHARS               27
#define ED_UNASSIGNED                    28
#define EM_CAPITOL_CASE                  29
#define EM_COPY_PREV_WORD                30
#define EM_COPY_REGION                   31
...
```

**Comment by lhoward** [comment]
Don't really see the harm in your patch as long as it doesn't break things for other platforms.

**Comment by quanah** [comment]
@lhoward It's specific to Solaris 10, thus why I put Solaris 10 in the title of the report. ;)

Solaris 10 is still in long term support for a while and we have customers using it.

**Comment by lhoward** [comment]
Can you send me the relevant snippet of `fcns.h` on your build?

**Comment by quanah** [comment]
odd, I reverted the patch and it's not reproducing. :/

**Comment by lhoward** [comment]
Are there build artefacts left? Did you try `make distclean`?

**Comment by quanah** [comment]
It was a completely clean build.

**Comment by lhoward** [comment]
What's your locale? Anyway, if it's no longer an issue feel free to close.

**Comment by quanah** [comment]
@lhoward I answered that in https://github.com/heimdal/heimdal/issues/662#issuecomment-561772842 :)

I wonder if one of the other solaris 10 fixes somehow addressed this problem, I'll close for now.
